### PR TITLE
rough splash screen

### PR DESCRIPTION
**Pull request by ljpenn9**

### CODE REVIEW

**Comment by TheHeroBrine422** [comment]
recommendation: update src/requirements.txt with whatever python packages you are using. Heres the syntax for the file: https://pip.pypa.io/en/stable/reference/requirements-file-format/

File: src/main.py [hunk 1]
```diff
@@ -1,13 +1,16 @@
-from networking import NetworkingManager
+#from networking import NetworkingManager
 from render import RenderingManager
-from database import DatabaseManager
+#from database import DatabaseManager
+from tkinter import *
```
Comment (TheHeroBrine422) [inline src/main.py]: Is there a reason you are importing tkinter to the main.py file?

File: src/render.py [hunk 2]
```diff
@@ -1,11 +1,77 @@
-class RenderingManager():
-    def __init__(self, gameState, networkingManager, databaseManager):
+import tkinter as tk
+from tkinter import *
+from tkinter import ttk
+from PIL import ImageTk, Image
```
Comment (TheHeroBrine422) [inline src/render.py]: Is there a reason you are importing PIL when its not being used? Same with some of the unused tkinter imports

File: src/render.py [hunk 3]
```diff
@@ -1,11 +1,77 @@
...
 
     def tick(self):
         # do actual rendering here.
         pass
 
+class SplashScreen(tk.Frame):
+    def __init__(self, parent, controller):
+        tk.Frame.__init__(self, parent)
+        # Add image file 
+        #SCREEN_WIDTH = parent.winfo_width()
```
Comment (TheHeroBrine422) [inline src/render.py]: is there a reason for these comments now? or are you going to remove the globals from the RenderingManager class?

File: src/render.py [hunk 4]
```diff
@@ -1,11 +1,77 @@
...
+        #SCREEN_HEIGHT = parent.winfo_height()
+        parent.bg = bg = PhotoImage(file = r'src/icons/logo.png')
+        parent.bg = bg = bg.subsample(2,2)
+        #parent.bg = bg = Image.open(r'src/icons/logo.png')
```
Comment (TheHeroBrine422) [inline src/render.py]: Is there a reason for these comments?

File: src/main.py [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
 
 gameState = {"running": True, "stage": "init"} # global stuff like what players on what team, what is the current state of the game (starting, inprogress, finished) and other stuff like that.
 
-databaseManager = DatabaseManager(gameState)
-networkingManager = NetworkingManager(gameState)
-renderingManager = RenderingManager(gameState, networkingManager, databaseManager)
+#databaseManager = DatabaseManager(gameState)
+#networkingManager = NetworkingManager(gameState)
+renderingManager = RenderingManager(gameState)
+renderingManager.mainloop()
```
Comment (TheHeroBrine422) [inline src/main.py]: It seems like this mainloop call blocks the below networking loop/calls from running. Is it possible to add the networking call to this mainloop or render tkinter without using its mainloop, but instead calling the render frame function directly?

File: src/main.py [hunk 6]
```diff
@@ -1,13 +1,16 @@
...
+#from database import DatabaseManager
+from tkinter import *
+from tkinter import ttk
 
 gameState = {"running": True, "stage": "init"} # global stuff like what players on what team, what is the current state of the game (starting, inprogress, finished) and other stuff like that.
 
-databaseManager = DatabaseManager(gameState)
-networkingManager = NetworkingManager(gameState)
-renderingManager = RenderingManager(gameState, networkingManager, databaseManager)
+#databaseManager = DatabaseManager(gameState)
```
Comment (TheHeroBrine422) [inline src/main.py]: I would prefer to see these uncommented so they can be used (same with above imports for these classes) before merging to main.

**Review by TheHeroBrine422 — CHANGES_REQUESTED** [review]

File: src/render.py (hunk 2 above)
Comment (TheHeroBrine422) [reply src/render.py]: this was partially addressed via discord. PIL was added for potential usage later. That is fine but I would like to see it added to requirements.txt then.

File: src/render.py [hunk 7]
```diff
@@ -1,11 +1,77 @@
...
+
+        # Using a method to switch frames
+        self.show_frame(SplashScreen)
+
+    def show_frame(self, cont):
+        frame = self.frames[cont]
+        # raises the current frame to the top
+        frame.tkraise()
 
     def tick(self):
```
Comment (TheHeroBrine422) [inline src/render.py:47]: this function should probably be removed if tkinter doesnt have a option for rendering single frames from inside of its tk.Tk class.

**Comment by TheHeroBrine422** [comment]
You might also want to move the SplashScreen class to its own file (something like src/screens/SplashScreen.py) and import it into the rendering manager to make working on multiple screens at once easier.

File: src/main.py (hunk 1 above)
Comment (ljpenn9) [reply src/main.py]: Thought I would need it to be using the mainloop() function, but it's being imported through RenderingManager, so I'll delete it.

File: src/render.py (hunk 4 above)
Comment (ljpenn9) [reply src/render.py]: Just old code. Will be removed.

File: src/render.py (hunk 3 above)
Comment (ljpenn9) [reply src/render.py]: Old code that will be removed

File: src/main.py (hunk 5 above)
Comment (ljpenn9) [reply src/main.py]: should be fixed by calling update() now instead of mainloop()

File: src/render.py (hunk 7 above)
Comment (ljpenn9) [reply src/render.py:47]: If tk.Tk doesn't, then we will need to **keep** this function. I couldn't find any documentation saying it does, so I'm leaving this function.

**Pull request merged**